src/mono/mono/mini/monovm.c

```diff
@@ -208,6 +210,8 @@ monovm_initialize (int propertyCount, const char **propertyKeys, const char **pr
 		return 0x80004005; /* E_FAIL */
 
 	finish_initialization ();
+		
+	mono_components_init ();
```

Comment:
1. If you make this change here, you also need to change `monovm_initialize_preparsed`.
2. But actually this will be bad for Xamarin.Android - they call `monovm_initialize_preparsed` very early.  Before they call `mono_dl_fallback_register` https://github.com/xamarin/xamarin-android/blob/0f7a0cde1425c29c52789e22192d1cc4c7af2334/src/monodroid/jni/monodroid-glue.cc#L1977 so component loading won't work.

Reply:
I will try to find another place to initialize it.